kate/recovery/src/data.rs

```diff
@@ -34,6 +34,26 @@ impl Cell {
 	}
 }
 
+/// Merges cells data per row.
+/// Cells are sorted before merge.
+pub fn rows(cells: &[Cell]) -> Vec<(u32, Vec<u8>)> {
+	let mut sorted_cells = cells.to_vec();
```

Comment:
If 1st step is `cells.to_vec()`, let's simplify the interface:
```suggestion
pub fn rows(mut cells: Vec<Cell>) -> Vec<(u32, Vec<u8>)> {
```

Reply:
But this mutates argument just for the sake of sorting (and we cannot enforce sorted vector through type I guess), its kinda awkward to have interface like that. Wdyt?

Reply:
See [this](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=c5a8e6a93010bc50d3d3624b6a21f2b1)

Reply:
@MiguelDD1 doesn't this change make the interface more ambiguous in terms whether the data is actually changing or not?

Reply:
I'm not sure about this one, by marking it as `mut` we allow caller to clone if needed (or to avoid cloning otherwise), but it introduces implicit coupling. If we remove sorting as some point, we need to change interface of fn. Also, its not clear why cells are mutated. I know its not that important here, but wanted to here more arguments.